XiEditor/Document.swift

```diff
@@ -224,4 +240,33 @@ class Document: NSDocument {
             editVC.updateAsync(update: update)
         }
     }
+    
+    /// Returns the frame to be used for the next new window.
+    /// - Note: This attempts to replicate the behaviour of native macOS applications.
+    /// On launch, an initial location is chosen, generally based on the position of
+    /// the last manually moved view during the application's last run; subsequent
+    /// windows are offset from this by approximately thie width of the title bar.
+    /// If a window clips the screen, the position starts again from the beggining
+    /// of the clipped axis.
+    func frameForNewWindow() -> NSRect {
+        let offsetSize: CGFloat = 22
+        let screenBounds = NSScreen.main!.visibleFrame
+        var nextFrame = NSOffsetRect(Document._lastWindowFrame, offsetSize, -offsetSize)
+        if nextFrame.maxX > screenBounds.maxX {
+            nextFrame.origin.x = screenBounds.minX + offsetSize
+        }
+        if nextFrame.minY <= 0 {
+            nextFrame.origin.y = screenBounds.maxY - nextFrame.height
+        }
```

Comment:
Might want to ensure that the new frame is not offscreen in the other direction either.

Reply:
A user can't drag a window above the menu bar unless they're dragging it to a separate view. Since we only move down and to the right, and wrap around to an edge when necessary, it shouldn't be possible for us to position a view out of bounds; the only exception would be if the user manually positions a view out so that it is clipped by the left edge of the screen before closing the app. In this case new windows will still progressively be moving into view.

So yea, maybe a manual check on first load that makes sure the left edge of the window is in bounds would make sense. :)